### PR TITLE
[sample_decode]P010 shift value is corrected

### LINKED ISSUE #1344: Regression: sample_decode fails for 10-bit HEVC when output size or format is specified
Setup:
- KBL
- Ubuntu 18.04
- Yesterday git version of drm-tip kernel
- Yesterday git versions of  MediaSDK and its dependencies

Test-case:
- `sample_multi_transcode -i::h265 crowd_run_1920x1080_50p_yuv420p10le.h265 -o::h265 output.h265 -w 300 -h 300 -hw`
- `sample_decode h265 -i crowd_run_1920x1080_50p_yuv420p10le.h265 -w 300 -h 300 -hw`

Expected outcome:
- sample_decode works, as sample_multi_transcode with same input & output size works

Actual outcome from sample_decode:
```
[ERROR], sts=MFX_ERR_UNSUPPORTED(-3), AllocFrames, m_pmfxVPP->Query failed at sample_decode/src/pipeline_decode.cpp:1078

[ERROR], sts=MFX_ERR_UNSUPPORTED(-3), Init, AllocFrames failed at sample_decode/src/pipeline_decode.cpp:442

[ERROR], sts=MFX_ERR_UNSUPPORTED(-3), main, Pipeline.Init failed at sample_decode/src/sample_decode.cpp:701
```
Notes:
- sample_decode works fine for 8-bit HEVC

### PR DESCRIPTION
**Pull request by fzhar**
Fixes #1344

### CODE REVIEW

**Comment by mgonchar** [comment]
@fzhar std::max issue is fixed in samples was merged as separate PR #1352 Could you please clean up commit history of this PR

File: samples/sample_common/include/brc_routines.h [hunk 1]
```diff
@@ -17,6 +17,7 @@ The original version of this sample may be obtained from https://software.intel.
 or https://software.intel.com/en-us/media-client-solutions-support.
 \**********************************************************************************/
 #include "sample_defs.h"
+#include "algorithm"
```
Comment (lakulako) [inline samples/sample_common/include/brc_routines.h]: Not needed for this commit.

Comment (onabiull) [reply samples/sample_common/include/brc_routines.h]: And should be in <>

Comment (onabiull) [reply samples/sample_common/include/brc_routines.h]: Please remove this line

**Pull request merged**